types/rollup/index.d.ts

```diff
@@ -138,9 +138,9 @@ export interface Plugin {
 	 */
 	resolveId?(importee: string, importer: string | undefined): string | null | undefined | false | 0 | ''
 	/** A module transformer function */
-	transform?(this: TransformContext, source: string, id: string): string | null | undefined | { code: string, map: SourceMap }
+	transform?(this: TransformContext, source: string, id: string): TransformResult | Promise<TransformResult>
```

Comment:
These seems to retune only a promise

Reply:
Check this as a guidance https://github.com/rollup/rollup/blob/master/src/rollup/index.ts

Reply:
But there is still a possibility to return instant values from these hooks. Moreover the official [plugin guidance](https://github.com/rollup/rollup/wiki/Plugins#creating-plugins) (maybe outdated) says no word about promises. In [official tests](https://github.com/rollup/rollup/blob/master/test/form/samples/absolute-path-resolver/_config.js) we can see transform hooks without promises.

Reply:
That is right, thanks for the clariffication.